test/test_red_black_tree.jl

```diff
@@ -0,0 +1,77 @@
+include("../src/red_black_tree.jl")
+@testset "RBTree" begin
+    t = RBTree{Int}()
+    for i = 1:10000
+        insert!(t, i)
+    end
+    for i = 1:10000
+        @test search_key(t, i)
+    end
+
+    for i = 10001:20000
```

Comment:
Why are these numbers so large?
Seems like it will not improve our actually test coverage and will make the tests slower and harder to read if they fail?